parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -171,8 +171,47 @@ CURL_OUTPUT=/tmp/curl_verbose.out
 ORAS_OUTPUT=/tmp/oras_verbose.out
 ORAS_REGISTRY_CONFIG_FILE=/etc/oras/config.yaml # oras registry auth config file, not used, but have to define to avoid error "Error: failed to get user home directory: $HOME is not defined" 
 
+# Checks if the elapsed time since CSEStartTime exceeds a specified maximum duration or the default 13 minutes.
+# Default value is based on the global CSE timeout of 15 minutes.
+# functions that use this should handle exit code 1 from this function to gracefully handle a "global" cse timeout (mean to avoid random 124 exit codes with no context)
+check_cse_timeout() {
+    shouldLog=${1:-true}
+    maxDurationSeconds=780 # 780 seconds = 13 minutes
+
+    if [[ -z "$CSE_STARTTIME_FORMATTED" ]]; then
```

Comment:
again, new SC rules incoming:

```suggestion
    if [[-z "$CSE_STARTTIME_FORMATTED" ]; then
```